src/interactive-auth.ts

```diff
@@ -159,7 +158,7 @@ export class NoAuthFlowFoundError extends Error {
  *
  * The generic parameter `T` is the type of the response of the endpoint, once it is eventually successful.
  */
-export type UIAuthCallback<T> = (makeRequest: (authData: AuthDict | null) => Promise<UIAResponse<T>>) => Promise<T>;
+export type UIAuthCallback<T> = (makeRequest: (authData: AuthDict | null) => Promise<T>) => Promise<T>;
```

Comment:
It would be nice, while you're here (and since it took me a while to figure out), to clarify the documentation to say that, in the case of more auth being needed, `makeRequest` will throw a `MatrixError` whose `data` conforms to  `IAuthData`. But I don't mind if you want to leave that for another time.